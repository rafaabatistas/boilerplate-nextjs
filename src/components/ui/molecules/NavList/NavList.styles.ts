import styled from 'styled-components';

export const NavList = styled.ul`
  width: 100%;
  height: 100%;
  max-width: 1240px;
  list-style: none;
  display: flex;
  justify-content: space-between;
  padding: 0 50px;
`;

export const Item = styled.li`
  display: block;
  width: 60px;
  height: 100%;
  display: flex;
  align-items: center;
`;

export const Logotipo = styled.img`
  width: 50px;
`;
