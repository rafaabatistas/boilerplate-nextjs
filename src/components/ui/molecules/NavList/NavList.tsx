import * as S from './NavList.styles';

import { ButtonLink } from '~atoms/ButtonLink/ButtonLink';

export const NavList = () => {
  return (
    <S.NavList>
      <S.Item>
        <S.Logotipo src="assets/img/icon-react.png" alt="Imagem escrito nextJS" />
      </S.Item>
      <S.Item>
        <ButtonLink link={'https://github.com/rafaabatistas/boilerplate-nextjs'}>GitHub</ButtonLink>
      </S.Item>
    </S.NavList>
  );
};
