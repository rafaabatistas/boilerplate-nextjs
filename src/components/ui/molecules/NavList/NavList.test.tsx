import { render } from '@testing-library/react';

import { NavList } from './NavList';

describe('<NavList />', () => {
  it('Deve renderizar o componente <NavList />', () => {
    //Arrange
    const { container } = render(<NavList />);
    //Assert
    expect(container.firstChild).toMatchSnapshot();
  });
});
