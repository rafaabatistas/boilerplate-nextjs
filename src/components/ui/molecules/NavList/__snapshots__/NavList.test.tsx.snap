// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavList /> Deve renderizar o componente <NavList /> 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  max-width: 1240px;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 50px;
}

.c1 {
  display: block;
  width: 60px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 50px;
}

.c3 {
  color: white;
  font-size: 1.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.5s ease-out;
  transition: all 0.5s ease-out;
}

.c3:hover {
  color: #adadad;
}

<ul
  class="c0"
>
  <li
    class="c1"
  >
    <img
      alt="Imagem escrito nextJS"
      class="c2"
      src="assets/img/icon-react.png"
    />
  </li>
  <li
    class="c1"
  >
    <a
      class="c3"
      href="https://github.com/rafaabatistas/boilerplate-nextjs"
      rel="noreferrer noopener"
      target="_blank"
    >
      GitHub
    </a>
  </li>
</ul>
`;
